DB/src/main/java/io/deephaven/db/v2/locations/parquet/topage/ToDBDateTimePage.java

```diff
@@ -3,39 +3,77 @@
 import io.deephaven.db.tables.dbarrays.DbArray;
 import io.deephaven.db.tables.dbarrays.DbArrayDirect;
 import io.deephaven.db.tables.utils.DBDateTime;
+import io.deephaven.db.tables.utils.DBTimeUtils;
 import io.deephaven.db.v2.sources.chunk.Attributes;
+import org.apache.parquet.schema.LogicalTypeAnnotation;
 import org.jetbrains.annotations.NotNull;
 
-public class ToDBDateTimePage<ATTR extends Attributes.Any> extends ToLongPage<ATTR> {
+import java.util.function.LongUnaryOperator;
 
-    private static ToDBDateTimePage INSTANCE = new ToDBDateTimePage<>();
+public abstract class ToDBDateTimePage<ATTR extends Attributes.Any> extends ToLongPage<ATTR> {
 
-    public static <ATTR extends Attributes.Any> ToDBDateTimePage<ATTR> create(@NotNull Class<?> nativeType) {
+    private static <ATTR extends Attributes.Any> ToDBDateTimePage<ATTR> fromUnitToNanos(final LongUnaryOperator convertUnitToNanos) {
+        return new ToDBDateTimePage<ATTR>() {
+            private long toNanos(final long v) {
+                return convertUnitToNanos.applyAsLong(v);
+            }
+
+            @Override
+            public final long[] convertResult(@NotNull final Object result) {
+                final long[] resultLongs = (long[]) result;
+
+                final int resultLength = resultLongs.length;
+                for (int ri = 0; ri < resultLength; ++ri) {
+                    resultLongs[ri] = toNanos(resultLongs[ri]);
+                }
+
+                return resultLongs;
+            }
+
+            @Override
+            @NotNull
+            public DbArray<DBDateTime> makeDbArray(long[] result) {
+                DBDateTime[] to = new DBDateTime[result.length];
+
+                for (int i = 0; i < result.length; ++i) {
+                    to[i] = new DBDateTime(toNanos(result[i]));
```

Comment:
This is wrong. You aren't mapping null times to the null `DBDateTime` correctly. The previous impl was correct.